lib/realtime/server.ex

```diff
@@ -379,14 +384,20 @@ defmodule Realtime.Server do
       _ = :ets.insert(ets, {{:route_id, route_id}, active_vehicles_and_ghosts})
     end
 
-    all_vehicles = Enum.uniq(all_vehicles(vehicles_by_route_id) ++ shuttles)
+    {logged_out_vehicles, logged_in_vehicles} =
+      vehicles_by_route_id
+      |> all_vehicles()
+      |> Enum.concat(shuttles)
+      |> Enum.uniq()
```

Comment:
**question:** I know your code wasn't the first to introduce this, but I wonder if it would be more efficient to do `uniq_by/2` here and compare the vehicles by IDs, rather than the whole maps.

More broadly, I want to better understand why the `uniq` step is necessary - my understanding is that that's because `all_vehicles` takes a map of routes to vehicles and combines them, and due to incoming vehicles a vehicle may be associated with multiple routes, but let me know if I'm wrong!

Reply:
I think your'e right that this is due to incoming vehicles which may be associated with multiple routes. If the vehicle structs for incoming vehicles are exactly equal to the regular vehicle than I agree `uniq_by` seems appropriate here! Everything I've seen so far in code suggests that they would be exactly equal. And if that were not the case, then searching for a vehicle that is in the incoming box would return two search results, which doesn't seem to be the case. 

There is [one test ](https://github.com/mbta/skate/blob/7d882b404c493ce651138451e7b7fdafedce2964/test/realtime/server_test.exs#L279)that looks for these values to be deduplicated that is testing with exactly duplicated shuttles. I haven't seen anything to point to there being a reason for duplicated shuttle vehicles with the same ID but different other properties.

With all that in mind, I can update to use `uniq_by` instead!